app/controllers/unattended_controller.rb

```diff
@@ -111,13 +111,12 @@ def render_default_global_template
   def render_local_boot_template
     return unless verify_found_host
 
-    ipxe_template_kind = TemplateKind.find_by(name: 'iPXE')
-    name = @host.local_boot_template_name(:iPXE) || ProvisioningTemplate.local_boot_name(:iPXE)
-    template = ProvisioningTemplate.find_by(name: name, template_kind: ipxe_template_kind)
+    name = ProvisioningTemplate.local_boot_template_name_for('iPXE')
+    template = ProvisioningTemplate.find_global_default_template(name, 'iPXE')
 
```

Comment:
Thank you, generally I understand the problem now and I agree with the solution, except one thing. Previously, the method:

```ruby
    def local_boot_template_name(kind)
      key = "local_boot_#{kind}"
      host_params[key] || host_params[key.downcase] || Setting[key]
    end
```

allowed users to overload the template name in host/hostgroup params. With this proposal this is gone. Can you retain the functionality? When not defined, fallback to the behavior you figured out does not work. I think the correct way to implement this is simply changing the `local_boot_template_name`:

```
host_params[key] || host_params[key.downcase] || Setting[ProvisioningTemplate.local_boot_name(key]]
```

Now it should either return overloaded template name OR correctly the "KIND default local boot" name which is indeed a valid name.

Reply:
Revised to accept a host parameter and if not present check/utilize value from settings.  Tested locally and believe this maintains the functionality you are describing while adding the missing functionality.  I missed the host param support in the docs so wasn't concerned with maintaining it when initially developing the patch.

Reply:
Apologies, holidays... Yes I think you got it right.